src/Microsoft.DotNet.Interactive/KeyValueStoreKernel.cs

```diff
@@ -181,19 +181,34 @@ void UndoSetValue()
         }
     }
 
-    private void StoreValue(
+    private async Task StoreValueAsync(
         string value,
         ValueDirectiveOptions options,
         KernelInvocationContext context,
         string mimeType = null)
+    {
+        mimeType ??= (options.MimeType ?? PlainTextFormatter.MimeType);
+
+        var shouldDisplayValue = options.MimeType is { } displayMimeType;
+
+        await StoreValueAsync(options.Name, value, mimeType, shouldDisplayValue, context);
+    }
+
+    protected virtual Task StoreValueAsync(
+        string key,
+        string value,
+        string mimeType,
+        bool shouldDisplayValue,
+        KernelInvocationContext context)
 
     {
-        mimeType ??= options.MimeType ?? PlainTextFormatter.MimeType;
-        _values[options.Name] = new FormattedValue(mimeType, value);
+        _values[key] = new FormattedValue(mimeType, value);
 
-        if (options.MimeType is {} displayMimeType)
+        if (shouldDisplayValue)
         {
-            context.DisplayAs(value, displayMimeType);
+            context.DisplayAs(value, mimeType);
         }
+
+        return Task.CompletedTask;
```

Comment:
Can the method be made sync or is this async because we think we may need it down the line?